app/models/agreements/iaa_gtc.rb

```diff
@@ -13,8 +13,22 @@ class Agreements::IaaGtc < ApplicationRecord
   validates :estimated_amount, numericality: { less_than: 10_000_000_000,
                                                greater_than_or_equal_to: 0,
                                                allow_nil: true }
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :end_date_after_start_date
 
-  def partner_status
-    iaa_status.partner_name
+  def status
+    return 'pending start' if Time.zone.today < start_date
+    return 'expired' if Time.zone.today > end_date
+    'active'
```

Comment:
are these new enums? I feel like `snake_case` enums are more common in APIs and reports

Reply:
You know that's a really good point. Until now I've mostly been thinking of this from a human-readable perspective but as an API snake case would make way more sense. I'll update, thanks!